### PR TITLE
chore: deprecate client options presence

### PR DESCRIPTION
**Pull request by didinele**

### CODE REVIEW

**Review by kyranet — APPROVED** [review]

**Comment by kyranet** [comment]
One little thing, you should mark this field as deprecated in the typings:
https://github.com/discordjs/discord.js/blob/5eabec14d45ef7bdd7f610e84234eb63e726eacd/packages/discord.js/typings/index.d.ts#L5307

**Review by Jiralite — APPROVED** [review]

**Comment by Jiralite** [comment]
Maybe deprecate the specific option in the typings too?

**Review by vladfrangu — DISMISSED** [review]
Requesting changes to the PR so the typing change is applied

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -5304,6 +5304,7 @@ export interface ClientOptions {
   allowedMentions?: MessageMentionOptions;
   partials?: readonly Partials[];
   failIfNotExists?: boolean;
+  /** @deprecated In the next release, use {@link ClientOptions.ws.initialPresence} instead */
```
Comment (kyranet) [inline packages/discord.js/typings/index.d.ts:5307]: By the time users see this, it'll be already that release:
```suggestion
  /** @deprecated Use {@link ClientOptions.ws.initialPresence} instead */
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: packages/discord.js/typings/index.d.ts (hunk 1 above)
Comment (didinele) [reply packages/discord.js/typings/index.d.ts:5307]: > By the time users see this, it'll be already that release:

No...? Vlad said we're deprecating in 14 and removing completely in 15.

Comment (didinele) [reply packages/discord.js/typings/index.d.ts:5307]: In retrospect though, wsOptions.presence is literally not a thing right now. It's probably best if we started supporting it in 14 right now.

Comment (Jiralite) [reply packages/discord.js/typings/index.d.ts:5307]: This is landing in the next version 14 release. The message they'll see is "In the next release use...", but it should be only "Use".

File: packages/discord.js/src/client/Client.js [hunk 2]
```diff
@@ -547,6 +557,12 @@ class Client extends BaseClient {
     if (typeof options.ws !== 'object' || options.ws === null) {
       throw new DiscordjsTypeError(ErrorCodes.ClientInvalidOption, 'ws', 'an object');
     }
+    if (
+      (typeof options.ws === 'object' && typeof options.ws.initialPresence !== 'object') ||
+      options.ws.initialPresence === null
```
Comment (almeidx) [inline packages/discord.js/src/client/Client.js:562]:
```suggestion
      typeof options.ws === 'object' &&
      (typeof options.ws.initialPresence !== 'object' || options.ws.initialPresence === null)
```

**Review by almeidx — DISMISSED** [review]

**Review by kyranet — APPROVED** [review]

**Review by almeidx — APPROVED** [review]

**Pull request merged**